docs/design/proposals/metro-dr.md

```diff
@@ -0,0 +1,70 @@
+# Metro-HA Disaster Recovery
+
+## Summary
+
+In the case of metro DR, a single ceph cluster is used between multiple kubernetes
+clusters, For disaster recovery the applications from one kubernetes
+cluster will be moved to another kubernetes cluster (expected that
+namespace name and the pvc name remains same) as a single ceph cluster is used
+by all kubernetes cluster the cephcsi running on the cluster where
+failover/fallback is happening should be able to identify the underlying
+RBD/CephFS volume and reattach to the PV.
+
+## Architecture
+
+![metro-dr](metro-dr.png)
+
+### Design
+
+Currently, ceph-csi generates its unique ID for volume and stores the
+corresponding PV name and the unique ID mapping, It creates the backend volume
+with the unique ID and returns the encoded value which contains all the
+required information for other operations. For metro DR, the same volume needs
+to be used in the secondary Kubernetes cluster. In the secondary cluster, the
+cephcsi cannot identify the volume corresponding to the PVC. Based on the
+assumption that the Kubernetes namespace and the PVC name remain the same when
+admin is doing the failover/fallback cephcsi can use the PVC name and namespace
+name to identify the RBD image. For that cephcsi needs to make a few changes.
+
+### Pre-req
+
+* The PVC name and namespace need to be passed to the CSI driver as part of the
+  volume create request.
+* A new storage-class parameter to indicate that image name need to be derived
+  from PVC name and namespace name.
+* An unique ID in the cephcsi ceph cluster configmap
+
+In the case of the primary cluster as the PVC name and the namespace name
+should be sent in volume create request, the cephcsi will generate omap data
+and creates its request name (PV name) and unique ID mapping in omap
+`csi.volume.<uniqueID>.PVName` : `volume UUID` (this is required to identify
+the RBD image/cephFS subvolume belongs to the PV) and creates an RBD image with
+the name `csi-vol-<pvcname>-<namespacename>-<UUID>`.  When doing the failover,
+cephcsi in the secondary cluster will check the volume is already present, If
+the image is present cephcsi will add its request name(PV Name) mapping to the
+UUID (`csi.volume.<uniqueID>.PVName` : `volume UUID`). As ID passed in
+configmap is unique across all the cephcsi instances accessing the ceph cluster
+will help cephcsi to identify the volume.
+
```

Comment:
As discussed, add that retaining the same image OMap is useful to preserve and reuse the volumeContext, and hence better, than creating a separate image OMap per peer kubernetes cluster that is sharing this Ceph pool.